### PR TITLE
feat: add religion on contact profile

### PR DESCRIPTION
**Pull request by djaiss**
Screen Shot 2022-11-02 at 9 34 57 AM

### CODE REVIEW

File: app/Domains/Contact/ManageReligion/Services/UpdateReligion.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        ]);
+
+        return $this->contact;
+    }
+
+    private function updateLastEditedDate(): void
+    {
+        $this->contact->last_updated_at = Carbon::now();
+        $this->contact->save();
+    }
```
Comment (asbiin) [inline app/Domains/Contact/ManageReligion/Services/UpdateReligion.php:78]: There is laravel built-in way to do this: https://laravel.com/docs/9.x/eloquent-relationships#touching-parent-timestamps

File: app/Domains/Contact/ManageReligion/Web/ViewHelpers/ModuleReligionViewHelper.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            ],
+        ];
+    }
+
+    public static function list(Contact $contact): Collection
+    {
+        $collection = $contact->vault->account->religions()->orderBy('position', 'asc')
+            ->get()->map(function ($religion) use ($contact) {
+                return self::dto($religion, $contact);
+            });
```
Comment (asbiin) [inline app/Domains/Contact/ManageReligion/Web/ViewHelpers/ModuleReligionViewHelper.php]: you can write

```suggestion
        $collection = $contact->vault->account->religions()->orderBy('position', 'asc')
            ->get()->map(fn ($religion) => self::dto($religion, $contact));
```

File: app/Domains/Contact/ManageReligion/Web/ViewHelpers/ModuleReligionViewHelper.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+        return $collection;
```
Comment (asbiin) [inline app/Domains/Contact/ManageReligion/Web/ViewHelpers/ModuleReligionViewHelper.php]: You can skip the variable assignment...

**Review by asbiin — CHANGES_REQUESTED** [review]
Cool feature

File: app/Domains/Contact/ManageReligion/Services/UpdateReligion.php (hunk 1 above)
Comment (djaiss) [reply app/Domains/Contact/ManageReligion/Services/UpdateReligion.php:78]: Ok so I didn't know and it's great.
However in our codebase, there are some time where contact is not a parent model and we still need to indicate that the contact's `updated_at` field should be updated.
This method from Laravel is a bit too _magical_ for me. I prefer to call it explicitly in the service so I know that this method explicitly needs to update the parent model.

File: app/Domains/Contact/ManageReligion/Web/ViewHelpers/ModuleReligionViewHelper.php (hunk 3 above)
Comment (djaiss) [reply app/Domains/Contact/ManageReligion/Web/ViewHelpers/ModuleReligionViewHelper.php]: Done

**Pull request merged**